repo2docker/buildpacks/python/__init__.py

```diff
@@ -17,23 +26,42 @@ def python_version(self):
 
         name, version, _ = self.runtime
 
-        if name != "python" or not version:
-            # Either not specified, or not a Python runtime (e.g. R, which subclasses this)
+        if name is not None and name != "python":
+            # Either not a Python runtime (e.g. R, which subclasses this)
             # use the default Python
             self._python_version = self.major_pythons["3"]
             self.log.warning(
                 f"Python version unspecified, using current default Python version {self._python_version}. This will change in the future."
             )
             return self._python_version
 
-        py_version_info = version.split(".")
-        py_version = ""
-        if len(py_version_info) == 1:
-            py_version = self.major_pythons[py_version_info[0]]
+        if name is None or version is None:
+            self._python_version = self.major_pythons["3"]
+            runtime_version = Version(self.major_pythons["3"])
+            self.log.warning(
```

Comment:
If we decide to go with my suggestion below, we should perhaps only show this warning if we don't end up picking a version based on pyproject.toml.